### PR TITLE
dpg, add content type as header parameter when body is optional and content type is single

### PR DESCRIPTION
**Pull request by haolingdong-msft**
Fix https://github.com/Azure/autorest.java/issues/1534

### CODE REVIEW

File: javagen/src/main/java/com/azure/autorest/mapper/ProxyMethodMapper.java [hunk 1]
```diff
@@ -204,6 +204,14 @@ public Map<Request, List<ProxyMethod>> map(Operation operation) {
             List<ProxyMethodParameter> parameters = new ArrayList<>();
             List<ProxyMethodParameter> allParameters = new ArrayList<>();
             List<ProxyMethod> proxyMethods = new ArrayList<>();
+            // add content-type parameter to allParameters when body is optional and there is single content type
+            boolean isBodyParamRequired = request.getParameters()
+                    .stream().filter(p -> p.getProtocol() != null && p.getProtocol().getHttp() != null ? p.getProtocol().getHttp().getIn() == RequestParameterLocation.BODY : false)
+                    .map(Parameter::isRequired).findFirst().orElse(false);
```
Comment (weidongxu-microsoft) [inline javagen/src/main/java/com/azure/autorest/mapper/ProxyMethodMapper.java:210]: nit, move this (and the `MethodUtil.getContentTypeCount(operation.getRequests()) == 1` check) after the if block of `settings.isDataPlaneClient()`.

reason, if some variable only useful for DPG, avoid it exposed to code not really need to see that var).

File: javagen/src/main/java/com/azure/autorest/util/MethodUtil.java [hunk 2]
```diff
@@ -103,6 +102,64 @@ public static Request tryMergeBinaryRequests(List<Request> requests, Operation o
         return selectedRequest;
     }
 
+    /**
+     * @param request   the request to put contentType on
+     * @param operation
+     * @return the created content type parameter
+     */
+    public static Parameter createContentTypeParameter(Request request, Operation operation) {
```
Comment (weidongxu-microsoft) [inline javagen/src/main/java/com/azure/autorest/util/MethodUtil.java:110]: Is there any change from the one that delete?

File: javagen/src/main/java/com/azure/autorest/util/MethodUtil.java [hunk 3]
```diff
@@ -242,13 +237,16 @@ private static int findIndexForContentTypeParam(List<Parameter> parameters) {
...
         for (Request request : requests) {
-            for (String mediaType : request.getProtocol().getHttp().getMediaTypes()) {
-                ChoiceValue choiceValue = new ChoiceValue();
-                choiceValue.setValue(mediaType);
-                Language language = new Language();
-                language.setName(mediaType.toUpperCase(Locale.ROOT));
-                language.setDescription("Content Type " + mediaType);
-                choiceValues.add(choiceValue);
+            if (request.getProtocol().getHttp().getMediaTypes() != null
+                    && !request.getProtocol().getHttp().getMediaTypes().isEmpty()) {
```
Comment (weidongxu-microsoft) [inline javagen/src/main/java/com/azure/autorest/util/MethodUtil.java:241]: nit, `CoreUtil.isNullOrEmpty`

File: javagen/src/main/java/com/azure/autorest/util/MethodUtil.java (hunk 2 above)
Comment (haolingdong-msft) [reply javagen/src/main/java/com/azure/autorest/util/MethodUtil.java:110]: no change

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**